File: 2017/problem_d/problem_d.cpp
#include <iostream>
#include <map>
#include <string>


int main() {
	std::map<std::string, int> snow_incr;
	snow_incr["heavy snowing"] = 10;
	snow_incr["snowing"] = 5;
	snow_incr["light rain"] = -3;
	snow_incr["heavy rain"] = -8;
	snow_incr["cloudy"] = 0;
	snow_incr["sunshine"] = -1;
	int snow_height = 0;
	std::string current_line;
	while(1) {
		std::getline(std::cin, current_line);
		if (current_line.compare("end") == 0)
			break;
		if (snow_incr.find(current_line) == snow_incr.end())
			return 1;
		snow_height += snow_incr[current_line];
		std::cout << snow_height << std::endl;
	}
	return 1;
}

File: 2017/problem_d/problem_d.py
snow_incr = {   "heavy snowing" : 10,
                "snowing" : 5,
                "light rain" : -3,
                "heavy rain" : -8,
                "cloudy" : 0,
                "sunshine" : -1}

curr_line = ""
snow_height = 0
while True:
    curr_line = input("")
    if curr_line == "end":
        break
    snow_height += snow_incr[curr_line]
    if snow_height < 0:
        snow_height = 0
    print(snow_height)